### PR TITLE
Add django auth framework

### PR DESCRIPTION
**Pull request by mbeacom**
##### Description
The goal of this PR is to enable the Django authentication and user framework, refactoring our current auth flow to use `User` handling with GH, and remove custom Github middleware / views.
##### Checklist
- [x] linter status: 100% pass
- [x] changes don't break existing behavior
- [x] commit message follows [commit guidelines](https://github.com/gitcoinco/web/blob/master/docs/CONTRIBUTING.md#step-4-commit)

##### Affected core subsystem(s)
Auth, Sessions

##### Testing
This will need tested on staging before being merged live.

##### Refers/Fixes
Fixes: #312 
Refs: #234

### CODE REVIEW

File: app/app/urls.py [hunk 1]
```diff
@@ -161,14 +162,14 @@
...
     url(r'^sitemap\.xml$', sitemap, {'sitemaps': sitemaps}, name='django.contrib.sitemaps.views.sitemap'),
-    # Github Integration
-    path('_github/', include('github.urls', namespace='github')),
     # Interests
     path('actions/bounty/<int:bounty_id>/interest/new/', dashboard.views.new_interest, name='express-interest'),
     path('actions/bounty/<int:bounty_id>/interest/remove/', dashboard.views.remove_interest, name='remove-interest'),
     path('actions/bounty/<int:bounty_id>/interest/', dashboard.views.interested_profiles, name='interested-profiles'),
     # Legacy Support
     path('legacy/', include('legacy.urls', namespace='legacy')),
+    re_path(r'^logout/$', auth_views.logout, name='logout'),
```
Comment (owocki) [inline app/app/urls.py]: we may have to modify some of the cloudfront settings associated with this

File: app/dashboard/views.py [hunk 2]
```diff
@@ -518,7 +537,7 @@ def bounty_details(request, ghuser='', ghrepo='', ghissue=0):
                 params['is_legacy'] = bounty.is_legacy  # TODO: Remove this following legacy contract sunset.
                 if profile_id:
                     profile_ids = list(params['interested_profiles'].values_list('profile_id', flat=True))
-                    params['profile_interested'] = request.session.get('profile_id') in profile_ids
+                    params['profile_interested'] = profile_id in profile_ids
```
Comment (owocki) [inline app/dashboard/views.py]: strangely enough it doesnt seem like `profile_interested` is even being used at all in the template

Comment (mbeacom) [reply app/dashboard/views.py]: Yeah, I noticed that but opted to not yank it out yet.  I can remove it here though.

Comment (owocki) [reply app/dashboard/views.py]: +1

**Comment by owocki** [comment]
added a new link to the django github url route in https://github.com/gitcoinco/web/pull/541 -- will need to make sure that is migrated in this PR!

**Comment by mbeacom** [comment]
@owocki Faucet GH URL reverse updated.

File: app/dashboard/views.py [hunk 3]
```diff
@@ -538,7 +544,21 @@ def bounty_details(request, ghuser='', ghrepo='', ghissue=0):
 
 
 def profile_helper(handle):
-    """Define the profile helper."""
+    """Define the profile helper.
```
Comment (jasonrhaas) [inline app/dashboard/views.py]: Docstrings 👌

**Comment by jasonrhaas** [comment]
Is there any additional testing that needs to be done?  I looked at the code but hard to get a feel for the new changes without running it through the bounty process.

**Comment by mkosowsk** [comment]
Will check for breadth tonight, will check for depth and test on the weekend :)

File: app/dashboard/views.py [hunk 4]
```diff
@@ -218,37 +235,48 @@ def uninterested(request, bounty_id, profile_id):
...
-    return JsonResponse({
-        'paginator': {
-            'num_pages': interests.paginator.num_pages,
-        },
-        'data': interests_data,
-    })
+        bounty = Bounty.objects.get(pk=bounty_id)
+    except Bounty.DoesNotExist:
+        return JsonResponse({'errors': ['Bounty doesn\'t exist!']},
+                            status=401)
```
Comment (mkosowsk) [inline app/dashboard/views.py]: Minor thing but if bounty doesn't exist this should be a 404, right?

File: app/dashboard/views.py [hunk 5]
```diff
@@ -218,37 +235,48 @@ def uninterested(request, bounty_id, profile_id):
...
+    try:
+        interest = Interest.objects.get(profile_id=profile_id, bounty=bounty)
+        bounty.interested.remove(interest)
+        maybe_market_to_slack(bounty, 'stop_work')
+        interest.delete()
+    except Interest.DoesNotExist:
+        return JsonResponse({
+            'errors': ['Party haven\'t expressed interest on this bounty.'],
+            'success': False},
+            status=401)
```
Comment (mkosowsk) [inline app/dashboard/views.py]: Same as above, 404 here?

**Comment by mkosowsk** [comment]
@mbeacom Also digging the naming convention 👍🏻raised very minor concerns over 401 vs 404 in a couple spots but otherwise LGTM :)

File: app/dashboard/views.py (hunk 4 above)
Comment (thelostone-mc) [reply app/dashboard/views.py]: Googled it up and came across [this](https://www.w3.org/Protocols/rfc2616/rfc2616-sec10.html#sec10.4.5)

Comment (mbeacom) [reply app/dashboard/views.py]: Not sure where/why this showed up here, but I didn't add that xD It was from existing code that changed lines.  Agreed, though!

**Comment by mkosowsk** [comment]
@mbeacom This guy just go live on production? 😍

Just a brainstorming question on DashboardTokensTest section... @thelostone-mc do you think we hit everything we needed to hit as far as assertions go for this?

File: app/dashboard/views.py (hunk 4 above)
Comment (thelostone-mc) [reply app/dashboard/views.py]: @mbeacom sure we believe you :P

Comment (mbeacom) [reply app/dashboard/views.py]: @thelostone-mc 😛  https://github.com/gitcoinco/web/blob/master/app/dashboard/views.py#L199

Comment (mkosowsk) [reply app/dashboard/views.py]: @thelostone-mc hmm you believe this or not? Not sure if I'm convinced 😂🤣

**Comment by thelostone-mc** [comment]
@mkosowsk It's on stage, tested it out
works fine ^_^

**Pull request merged**